examples/tokens/ERC1155ownable.vy

```diff
@@ -0,0 +1,358 @@
+# @version >=0.3.2
+
+# @dev Implementation of ERC-1155 non-fungible token standard
+# @dev ownable, with approval, OPENSEA compatible (name, symbol)
+# @author Dr. Pixel (github: @Doc-Pixel)
+
+############### imports ###############
+from vyper.interfaces import ERC165
+
+############### variables ###############
+# maximum items in a batch call. Set to 128, to be determined what the practical limits are.
+BATCH_SIZE: constant(uint256) = 128             
+
+# URI length set to 1024. 
+MAX_URI_LENGTH: constant(uint256) = 1024        
+
+# the contract owner
+owner: public(address)                          
+
+# pause status True / False
+# @dev not part of the core spec but a common feature for NFT projects
+paused: public(bool)                            
+
+# the contracts URI to find the metadata
+uri: public(String[MAX_URI_LENGTH])                     
```

Comment:
There should be a function named 'uri' returning the URI. by making the variable uri public we have that covered.
I can create a public function uri and use a private variable baseuri if you like. On the outside it will look exactly the same.

Reply:
Reading through the ERC: https://eips.ethereum.org/EIPS/eip-1155#metadata

Not sure if this implemented correctly. The function described by the ERC conforms to `def uri(tokenId: uint256) -> String`, where this public getter will look like `def uri() -> String` instead.

Typically, at least for ERC721 tokens, there is a `baseURI` variable that exists, and then you would form a token-specific URI by doing some sort of combination of the `baseURI` and `tokenId` such that you would obtain that value (e.g. `ipfs://QmAbC...123/1.json` or similar)

I think some further work is needed here

Reply:
You're right!

Reply:
reworked this
changed public variable uri to private _uri
added function `def uri(id: uint256) -> String[MAX_URI_LENGTH]:` and rewired everything to use the _uri variable.
adjusted tests to call uri(0)